Sources/Core/EthereumABI/ABIElements.swift

```diff
@@ -325,7 +325,7 @@ extension ABI.Element.Function {
             // set a flag to detect the request succeeded
         }
 
-        if returnArray.isEmpty {
+        if returnArray.isEmpty && !outputs.isEmpty && data.isEmpty{
```

Comment:
Is this already reflected in tests?

Reply:
I have now removed the data check.
if outputs is empty return array can be empty, however we still want to return success not nil